codice/ddf
# [2.13.x] DDF-4320 Fix CDM ingesting too soon on system start

#### What does this PR do?
- Defines a json file specifying InputTransformer IDs started by default in DDF
- CDM bundle will fail to start unless all configured InputTransformer IDs are satisfied. This will prevent the CDM ingesting poor metadata or failed ingests on system restarts.
- Cleans up embedded bundles

#### Who is reviewing it? 
@emmberk @kcover @brendan-hofmann @pvargas @mdang8

#### Select relevant component teams: 
@codice/io

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@tbatie

#### How should this be tested?
Set log:set TRACE `org.codice.ddf.catalog.transformer.bootflag` for detailed logs.
Verify the following scenarios.

- Start a normal CDM (move/delete/inplace) and verify it begins ingesting
- Start a normal CDM (move/delete/inplace). Restart the distribution and verify it does not start until all InputTransformers are available (view logs).
- Before a fresh install, edit the ddf-default-input-transformers.json file to include an InputTransformer ID that doesn't exist. Verify install fails.

#### Any background context you want to provide?

#### What are the relevant tickets?
[DDF-4320](https://codice.atlassian.net/browse/DDF-4320)

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
